### PR TITLE
Removing parameter LL

### LINKED ISSUE #31: Bucket parameters no longer need to be a linked list
Because part of the startup process is to get the number of parameters in a bucket, a linked list is no longer needed. Replace with a U16 array.

### PR DESCRIPTION
**Pull request by caljay98**
Replaced the parameter LL with an array because parameter lists are now constant sizes

### CODE REVIEW

**Comment by caljay98** [comment]
Tested and working with no changes

File: DLM_code/Src/dlm-manage_data_aquisition.c [hunk 1]
```diff
@@ -112,8 +112,24 @@ void set_bucket_size(U8 sending_dam, void* UNUSED,
         above_bucket_node->next = bucket_node;
     }
 
-    // set the number of parameters in this bucket
+    // set the number of parameters in this bucket. No parameters have been added yet
+    bucket_node->bucket.params_added = 0;
```
Comment (hocally) [inline DLM_code/Src/dlm-manage_data_aquisition.c:116]: Is this intended? The comment and the old line seem to indicate this shouldn't be zero, though I haven't read the while file yet lol

File: DLM_code/Src/dlm-manage_data_aquisition.c [hunk 2]
```diff
@@ -112,8 +112,24 @@ void set_bucket_size(U8 sending_dam, void* UNUSED,
...
     bucket_node->bucket.num_of_params = num_of_params;
+
+    // free any memory that may have been used for the old parameter array
+    free(bucket_node->bucket.param_ids);
+
+    // malloc some memory for the new parameter array
+    bucket_node->bucket.param_ids = (BUCKET_PARAM_INFO*)malloc(num_of_params * sizeof(BUCKET_PARAM_INFO));
```
Comment (hocally) [inline DLM_code/Src/dlm-manage_data_aquisition.c:123]: This malloc call and the check for a null pointer could into a single conditional, but I'd say it's personal preference on style. This way is pretty clear ngl

File: DLM_code/Src/dlm-manage_data_aquisition.c [hunk 3]
```diff
@@ -155,51 +171,37 @@ void add_param_to_bucket(U8 sending_dam, void* UNUSED,
...
+		send_can_command(PRIO_HIGH, sending_dam, BUCKET_OK, bucket_id, 0, 0, 0);
+	}
+
+    // get a pointer to the location of the parameter array
+    param_array = bucket_node->bucket.param_ids;
 
     // Check to make sure this parameter is not already in the list. Do not add it
     // if it is
-    while (param_node != NULL)
+    for (c = 0; c < bucket_node->bucket.params_added; c++)
```
Comment (hocally) [inline DLM_code/Src/dlm-manage_data_aquisition.c:186]: c isn't super descriptive, can't hurt to give this a clearer name

File: DLM_code/Src/dlm-manage_data_aquisition.c [hunk 4]
```diff
@@ -245,7 +247,8 @@ void request_all_buckets()
 {
 	// Skip the head node
     BUCKET_NODE* bucket_node = bucket_list_head.next;
-    U16_LIST_NODE* param_node;
+    BUCKET_PARAM_INFO* param_array;
+    U8 c;
```
Comment (hocally) [inline DLM_code/Src/dlm-manage_data_aquisition.c:251]: Why is this uninitialized?

Comment (hocally) [reply DLM_code/Src/dlm-manage_data_aquisition.c:251]: Wait nvm I think I see why. Same naming comment as above applies here

File: DLM_code/Src/dlm-manage_data_aquisition.c [hunk 5]
```diff
@@ -290,28 +293,29 @@ void store_new_data()
 {
 	// Skip the head node
     BUCKET_NODE* bucket_node = bucket_list_head.next;
-    U16_LIST_NODE* param_node;
+    BUCKET_PARAM_INFO* param_array;
     CAN_INFO_STRUCT* param_info;
+    U8 c;
```
Comment (hocally) [inline DLM_code/Src/dlm-manage_data_aquisition.c:298]: Naming comment from above

File: DLM_code/Src/dlm-manage_data_aquisition.c [hunk 6]
```diff
@@ -348,10 +352,10 @@ S8 add_param_to_ram(U16_LIST_NODE* param_node, BUCKET_NODE* bucket_node)
     DATA_INFO_NODE* data_node;
     CAN_INFO_STRUCT* can_param_struct;
 
-    can_param_struct = (CAN_INFO_STRUCT*)(all_parameter_structs[param_node->data]);
+    can_param_struct = (CAN_INFO_STRUCT*)(all_parameter_structs[param_info->parameter]);
 
     // Choose the correct type of data node based on the parameter data type, then malloc the memory needed
-    switch (parameter_data_types[param_node->data])
+    switch (parameter_data_types[param_info->parameter])
```
Comment (hocally) [inline DLM_code/Src/dlm-manage_data_aquisition.c:358]: Could just be because I'm on my phone but the indenting looks weird here

**Review by hocally — APPROVED** [review]

**Comment by hocally** [comment]
@caljay98 Looks good, great job testing the changes. Left some stylistic comments, feel free to address either now or mix into a subsequent PR

File: DLM_code/Src/dlm-manage_data_aquisition.c (hunk 1 above)
Comment (caljay98) [reply DLM_code/Src/dlm-manage_data_aquisition.c:116]: There are two values in the bucket, number of parameters currently added and number of parameters that need to be to make it complete

**Pull request merged**